### PR TITLE
Fetch Nimble via SPM and upgrade to version 10.0.0

### PR DESCRIPTION
**Pull request by mokagio**
What it says in the title. The upgrade from 9.x to 10.x ~~was straightforward for us~~ had limited footprint, so I felt it was okay to batch the two changes in one PR. See also #19079.

## Testing

If CI is green, the dependency manager migration and library upgrade was successful.

## Regression Notes
1. Potential unintended areas of impact – Some tests might break
2. What I did to test those areas of impact (or what existing automated tests I relied on) – See what CI does
3. What automated tests I added (or what prevented me from doing so) – N.A.

PR submission checklist:

- [x] I have completed the Regression Notes. – N.A.
- [x] I have considered adding unit tests for my changes. – N.A.
- [x] I have considered adding accessibility improvements for my changes. – N.A.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary. – N.A.

### CODE REVIEW

File: WordPress/WordPressTest/ViewRelated/Post/Controllers/PostListViewControllerTests.swift [hunk 1]
```diff
@@ -50,7 +50,7 @@ class PostListViewControllerTests: CoreDataTestCase {
 
         // Given
         let blog = BlogBuilder(mainContext).build()
-        try! mainContext.save()
+        try mainContext.save()
```
Comment (mokagio) [inline WordPress/WordPressTest/ViewRelated/Post/Controllers/PostListViewControllerTests.swift:53]: This is unrelated to the Nimble upgrade, but seemed appropriate to add here.

File: WordPress/WordPressTest/ViewRelated/Post/Controllers/PostListViewControllerTests.swift [hunk 2]
```diff
@@ -62,10 +62,15 @@ class PostListViewControllerTests: CoreDataTestCase {
...
+        // Note that `XCTAssertNoThrow` catches `NSException`s as well as Swift's `throw`.
+        //
+        // This test originally used Nimble's `raiseException` but that matcher is no longer available in the SPM build.
+        // See https://github.com/Quick/Nimble/blob/e313d9a67ec2e4171d416c61282e49fc3aadc7a4/Sources/Nimble/Matchers/RaisesException.swift#L1
+        XCTAssertNoThrow(try {
             _ = PostBuilder(self.mainContext, blog: blog).with(status: .draft).build()
-            try! self.mainContext.save()
-        }.notTo(raiseException())
+            try self.mainContext.save()
+        }())
```
Comment (mokagio) [inline WordPress/WordPressTest/ViewRelated/Post/Controllers/PostListViewControllerTests.swift:73]: Took me a while to figure this one out. From the commit message:

You can verify the change works by modifying the source code with the following diff:

```diff
--- a/WordPress/WordPressTest/PostBuilder.swift
+++ b/WordPress/WordPressTest/PostBuilder.swift
@@ -200,6 +201,7 @@ class PostBuilder {
     func build() -> Post {
         // TODO: Enable this assertion once we can ensure that the post's MOC isn't being deallocated after the `PostBuilder` is
         // assert(post.managedObjectContext != nil)
+        NSException(name: .genericException, reason: "testing").raise()
         return post
     }
```

You can also verify the `XCTAssertNoThrow` behavior in isolation with a test like the following:

```swift
func test_XCTAssertNoThrow_failsWhenExceptionRaised() throws {
    XCTExpectFailure("Expected failure to verify XCTAssertNoThrow fails when an exception raises")
    XCTAssertNoThrow(NSException.fixture().raise())
}
```

---

cc @shiki as the test author. Let me know if my edits seem sound.

**Review by crazytonyli — APPROVED** [review]

**Comment by mokagio** [comment]
I'm going to merge even though @shiki hasn't had a chance to comment yet. My changes are "just in the tests" which is not to say that we can be reckless with them, but we can be more lenient in merging and fixing issues later on.

File: WordPress/WordPressTest/ViewRelated/Post/Controllers/PostListViewControllerTests.swift (hunk 2 above)
Comment (shiki) [reply WordPress/WordPressTest/ViewRelated/Post/Controllers/PostListViewControllerTests.swift:73]: Looks okay to me. 🙂

**Pull request merged**